### PR TITLE
Adding a benchmarck in the distribution

### PR DESCRIPTION
**Pull request by huitema**
This is a first step, only measuring AEAD protocols, but it already gave me some interesting indications!

### CODE REVIEW

**Comment by huitema** [comment]
@kazuho I could add additional benchmarking for CTR and ECB modes, but this could go in a separate PR later.

File: t/ptlsbench.c [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+#ifdef _WINDOWS
+#define BENCH_OS "windows"
+#ifdef _DEBUG
+#define BENCH_MODE "check"
+#else
+#define BENCH_MODE "release"
+#endif
+#else
+#define BENCH_OS "linux"
```
Comment (kazuho) [inline t/ptlsbench.c]: Being surprised to see the OS reported when running the code on macOS, I think this should be "posix".

Comment (huitema) [reply t/ptlsbench.c]: Is there a defined name that I can check to differentiate posix, ios, debian, etc? How would you do that?

Comment (kazuho) [reply t/ptlsbench.c]: The proper way is to use the [uname](http://pubs.opengroup.org/onlinepubs/9699919799/functions/uname.html) function. Example: https://gist.github.com/kazuho/2d291582ac9ee76680447210adbc6982

Comment (kazuho) [reply t/ptlsbench.c]: And if you want to distinguish between linux versions, IIRC you should read `/etc/lsb-release`.

Comment (huitema) [reply t/ptlsbench.c]: I implemented the call to uname for unix systems. I dont think that the OS version matters very much. Instead, I added a display of the openssl version, on the theory that algorithm implementations may change with the SSL version. I would like to do the same for minicrypto, but I did not find a version number for the minicrypto library.

My goal is to be able to record various implementations in a single spreadsheet, which we could then tabulate with a variety of criteria. I am missing an indication of processor; maybe uname.machine would be useful.

I also added a display of the encryption/decryption speed in Mbps, and that's interesting.

**Pull request merged**